core/src/main/scripts/importer/validateData.py

```diff
@@ -3100,14 +3110,49 @@ def processCaseListDirectory(caseListDir, cancerStudyId, logger,
         else:
             stableid_files[stable_id] = case
 
-        sampleIds = meta_dictionary['case_list_ids']
-        sampleIds = set([x.strip() for x in sampleIds.split('\t')])
-        for value in sampleIds:
+        # Check if case list category is valid
+        VALID_CATEGORIES = ['all_cases_in_study',
+                            'all_cases_with_mutation_data',
+                            'all_cases_with_cna_data',
+                            'all_cases_with_log2_cna_data',
+                            'all_cases_with_methylation_data',
+                            'all_cases_with_mrna_array_data',
+                            'all_cases_with_mrna_rnaseq_data',
+                            'all_cases_with_rppa_data',
+                            'all_cases_with_microrna_data',
+                            'all_cases_with_mutation_and_cna_data',
+                            'all_cases_with_mutation_and_cna_and_mrna_data',
+                            'all_cases_with_gsva_data',
+                            'other']
+        if 'case_list_category' in meta_dictionary:
+            if meta_dictionary['case_list_category'] not in VALID_CATEGORIES:
+                logger.error('Invalid case list category',
+                               extra={'filename_': case,
+                               'cause': meta_dictionary['case_list_category']})
+
+
+        # Check for any duplicate sample IDs
+        sample_ids = [x.strip() for x in meta_dictionary['case_list_ids'].split('\t')]
+        seen_sample_ids = set()
+        dupl_sample_ids = set()
+        for sample_id in sample_ids:
+            if sample_id not in seen_sample_ids:
+                seen_sample_ids.add(sample_id)
+            else:
+                dupl_sample_ids.add(sample_id)
+        if len(dupl_sample_ids) > 0:
+            logger.warning('Duplicate Sample ID in case list',
```

Comment:
please specify in the message (or in the comments above it) why this is **not** an error. Is this because the sample will be filtered out by the loader?

Reply:
Done